### PR TITLE
offline mode for processing vectors

### PR DESCRIPTION
**Pull request by bfussell**
and a few cleanup items

Have not removed cURL from build, mURL can be used.
will create a spike to eval removing cURL.

steps need to be added to README  with decriptions -will create user story for that.
they are:

app/.libs/acvp_app --kas_ffc --vector_req req.json --none
app/.libs/acvp_app --all_algs --vector_req req.json  --vector_rsp rsp.json
app/.libs/acvp_app --all_algs --vector_upload rsp.json

### CODE REVIEW

File: include/acvp/acvp.h [hunk 1]
```diff
@@ -2513,6 +2513,17 @@ ACVP_RESULT acvp_set_certkey(ACVP_CTX *ctx, char *cert_file, char *key_file);
  */
 ACVP_RESULT acvp_mark_as_sample(ACVP_CTX *ctx);
 
+/*! @brief acvp_mark_as_request_only() marks the registration as a request only.
+
+    This function sets a flag that will allow the client to retrieve
+    the vectors from the server and store them in a file for later use.
+
+    @param ctx Pointer to ACVP_CTX that was previously created by
+        calling acvp_create_test_session.
```
Comment (fliphil) [inline include/acvp/acvp.h]: Missing param for filename here

File: include/acvp/acvp_lcl.h [hunk 2]
```diff
@@ -1253,6 +1253,9 @@ struct acvp_ctx_t {
     char *json_filename;
     int use_json;
     int is_sample;
+    char *vector_req_file;
```
Comment (fliphil) [inline include/acvp/acvp_lcl.h]: Comments for these fields may help :)

File: src/parson.c [hunk 3]
```diff
@@ -1650,6 +1650,65 @@ JSON_Status json_serialize_to_file_pretty(const JSON_Value *value, const char *f
...
 }
 
+JSON_Status json_serialize_to_file_pretty_a(const JSON_Value *value, const char *filename) {
+    JSON_Status return_code = JSONSuccess;
+    FILE *fp = NULL;
+    char *serialized_string = NULL; 
+
+    fp = fopen(filename, "a");
+    if (fp == NULL) {
+        json_free_serialized_string(serialized_string);
```
Comment (fliphil) [inline src/parson.c]: Dont need to free the string here right?

File: src/parson.c [hunk 4]
```diff
@@ -1650,6 +1650,65 @@ JSON_Status json_serialize_to_file_pretty(const JSON_Value *value, const char *f
...
+    json_free_serialized_string(serialized_string);
+    return return_code;
+}
+
+JSON_Status json_serialize_to_file_pretty_w(const JSON_Value *value, const char *filename) {
+    JSON_Status return_code = JSONSuccess;
+    FILE *fp = NULL;
+    char *serialized_string = json_serialize_to_string_pretty(value, NULL);
+    if (serialized_string == NULL) {
+        return JSONFailure;
```
Comment (fliphil) [inline src/parson.c]: Do need to free the string here before return

File: src/acvp.c [hunk 5]
```diff
@@ -803,6 +807,334 @@ ACVP_RESULT acvp_load_kat_filename(ACVP_CTX *ctx, const char *kat_filename) {
...
+        json_result = json_serialize_to_string_pretty(kat_val, NULL);
+        file_val = json_parse_string(json_result);
+        json_free_serialized_string(json_result);
+
+        /* track first vector set with file count */
+        if (n == 1) {
+            ACVP_STRING_LIST *vs_entry = NULL;
+
+            rsp_val = json_value_init_object();
+            rsp_obj = json_value_get_object(rsp_val);
```
Comment (fliphil) [inline src/acvp.c]: Can you not use the val/obj that you parsed earlier from the req file? Instead of making new ones here

File: src/acvp.c [hunk 6]
```diff
@@ -1528,15 +1891,52 @@ static ACVP_RESULT acvp_process_vsid(ACVP_CTX *ctx, char *vsid_url) {
...
+
+            /*
+             * Save the KAT VectorSet to file
+             */
+            if (ctx->vector_req) {
+                alg_array = json_value_get_array(val);
+                alg_val = json_array_get_value(alg_array, 1);
+
+                /* track first vector set with file count */
+                if (count == 0) {
```
Comment (fliphil) [inline src/acvp.c]: What is count being used for?

File: include/acvp/acvp_lcl.h (hunk 2 above)
Comment (bfussell) [reply include/acvp/acvp_lcl.h]: sure

File: src/parson.c (hunk 3 above)
Comment (bfussell) [reply src/parson.c]: correct, will remove

File: include/acvp/acvp.h (hunk 1 above)
Comment (bfussell) [reply include/acvp/acvp.h]: correct will add

File: src/acvp.c (hunk 5 above)
Comment (bfussell) [reply src/acvp.c]: I am trying to be very explicit there since there are several values used in this function

File: src/parson.c [hunk 7]
```diff
@@ -1650,6 +1650,65 @@ JSON_Status json_serialize_to_file_pretty(const JSON_Value *value, const char *f
...
+        }
+    }
+    if (fclose(fp) == EOF) {
+        return_code = JSONFailure;
+    }
+    json_free_serialized_string(serialized_string);
+    return return_code;
+}
+
+JSON_Status json_serialize_to_file_pretty_w(const JSON_Value *value, const char *filename) {
```
Comment (fliphil) [inline src/parson.c]: Unless you are using private functions here, might want to move these functions to utils.c instead. To try and keep this file as sane as possible for future syncs from the official

Comment (bfussell) [reply src/parson.c]: agree will do

**Pull request merged**